File: csci294/2020Spring/inclass/wk04_2020-02-03_mon/Function1.py
# Function1.py

def funA(counter):
    if counter > 0:
        print('FunA', counter)
        funB(counter - 1)    
        
def funB(counter):
    if counter > 0:
        print('FunB', counter)
        funA(counter - 1)

def main():
    funA(5)

main()

File: csci294/2020Spring/inclass/wk04_2020-02-03_mon/Function1.cpp
/* Function1.cpp */
// Demo need for function prototypes

#include <iostream>
using namespace std;

// Function Prototypes
void funA(int);
void funB(int);

int main(){
    funA(5);
}

void funA(int counter){
    if(counter > 0){
        cout << "funA " << counter << endl;
        funB(counter - 1);
    }
}

void funB(int counter) {
    if(counter > 0) {
        cout << "FunB " << counter << endl;
        funA(counter -1);
    }
}

